tools/python_api/src_cpp/py_connection.cpp

```diff
@@ -100,7 +124,7 @@ static std::unique_ptr<function::ScanReplacementData> replacePythonObject(
     }
     if (nameMatchFound) {
         throw BinderException(
-            stringFormat("Variable {} found but no matches were DataFrames", objectName));
+            stringFormat("Variable {} found but no matches were scannable", objectName));
```

Comment:
Could this error message be made a bit more helpful? Something like `Unable to scan {}. Can only scan from Pandas/Polars DataFrames or Arrow tables`. I feel like users should be immediately made aware of what works here rather than the fact that it doesn't work.

Reply:
I initially didn't do it because I thought it wouldn't scale well in the future, but it would probably work given that we already have much larger error messages. Perhaps we could tell the user the type of the variable that was found as well.

Reply:
Yup, I recommend something like `Found {} of type {} which is not supported for scanning`.